pkg/controller/hibernation/aws_actuator.go

```diff
@@ -0,0 +1,150 @@
+package hibernation
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awsclient "github.com/openshift/hive/pkg/awsclient"
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+// AWS machine states
+const (
+	instancePending      = 0
+	instanceRunning      = 16
+	instanceShuttingDown = 32
+	instanceTerminated   = 48
+	instanceStopping     = 64
+	instanceStopped      = 80
+)
+
+func init() {
+	RegisterActuator(&awsActuator{})
+}
+
+type awsActuator struct {
+}
+
+// CanHandle returns true if the actuator can handle a particular ClusterDeployment
+func (a *awsActuator) CanHandle(cd *hivev1.ClusterDeployment) bool {
```

Comment:
Love this change to the actuator pattern. In a future PR, we should probably change our other actuators to use this pattern.